lib/rubocop/cop/commissioner.rb

```diff
@@ -119,6 +121,18 @@ def with_cop_error_handling(cop, node = nil)
         err = ErrorWithAnalyzedFileLocation.new(cause: e, node: node, cop: cop)
         @errors << err
       end
+
+      def with_inline_config(cop, node = nil)
```

Comment:
That's the code that concerns me in terms of performance. We have to be really careful with any overhead added to the node callbacks.